### ISSUE TITLE
Trusted exchanges are DISABLING BYTECOIN

### ISSUE THREAD

**Issue by Ben19904** [amjuarez/bytecoin#127]
Following my original issue I posted ~2 weeks ago (https://github.com/amjuarez/bytecoin/issues/116), I have still had no luck in receiving my BCN.

As a quick summary, I bought bytecoin from changer.com (a trusted cryptocurrency exchange that I have used many times before), and saw that the transaction completed in chainradar. My bytecoin wallet is fully synced, but my BCN balance still appears a 0.00.

I tried many workarounds such as deleting the blockchain and re downloading from bytecoin.org, and importing that into the bytecoin wallet to no success. I have also tried using different versions of the bytecoin wallets, still to no success.

I am advanced in IT (I develop/maintain IT infrastructure for an investment bank) and so rest assured I don't make stupid mistakes such as the following:

  - Oops, I used the wrong wallet address!
  - I'm not synced with the network! DUH!
  - The transaction hasn't completed! DUH!
  - Oops, I deleted the .wallet file!
  - Oops, I forgot to backup the .wallet file!
  - Oops, I forgot to import the .wallet file backup into the new bytecoin wallet!

(so please don't make patronising suggestions as such :D)

After seeing other people are having similar issues to me (albeit - most most people's issues are due to one of the points as listed above - but there are others having real technical issues), and after speaking with changer.com support, it is clear that there is a serious problem with either the bytecoin network, or the bytecoin wallet.

After speaking with changer.com for a lengthy amount of time, they decided they would disable the buying of bytecoin from their website; see changer.com homepage, and notice that under the "receive" column, bytecoin is greyed out.

If I cannot get hold of these shady developers to recover my BCN, I will continue my destructive tantrum and encourage more exchanges to disabled bytecoin transactions, thus devaluing BCN further and further.

**Comment by boomworking** [comment]
may I help u contact me fb chaiwat boonsom

**Comment by Ben19904** [comment]
Sure - although you'll have to be more specific than that, there're loads of chaiwat boonsoms on FB :)

**Comment by SMH17** [comment]
The title of the report isn't appropriate and could create useless panic.  changer.com has temporary suspended the BCN purchase waiting the addressing of this issue since you and some other user have reported problem is way different than "trusted exchanges are disabling BCN" . The network is ok, wallet has some issue that needs a fix, you can try to use reference wallet code, but masochistic blackmail behavior "answer me or I convince the exchange to suspend" can create damage to the entire community, so report the issue and wait, Bytecoin isn't a regular corporation with customer care service and developers have more interest than you to fix everything.

**Comment by Ben19904** [comment]
I have already tried using the ref client ages ago and it didn't work.

Damaging the community... What community? There is no place you can go on the internet where you can talk to bytecoin techies - I have been trying for weeks.

I think that the community ought to know of this serious issue that has affected many, and if it scares people out of buying bytecoin, then so be it!

**Comment by SMH17** [comment]
If you want to invest in new altcoins still under heavy development you have to take in account some issue. If you want customer care assistance and no issue, use a credit card or PayPal not altcoins still under development. <<Trusted exchanges are DISABLING BYTECOIN >> is useless FUD, and a false claim, since only "one" exchange "seems" is considering to temporary disabled after you and some other have reported issue. If you think Bytecoin has no community why you have purchased it? My Bytecoin balance like the balance of many others users is ok, and considering that there are allocated 400 millions of dollars, the people with issue are certainly a small minority. 
If you look for the community go to discuss techies go to Cryptonote forum. Bytecoin is the reference Cryptonote altcoin. 
GitHub is a platform where developers collaborate sharing code and reporting issues to help the projects to grow. So if you want to report problems you are welcome, but try to be polite and constructive, describe the issue, the version of the software used, the date of creation of in/out transaction and your platform and wait, some time ago I have encountered a problem with the client that wasn't able to load correctly my balance and I have fixed the issue rebuilding the local db that seems the main culprit of these kind of errors. If you have a clean installation of latest version and you still get the issue, You can try to use the source to built a wallet that doesn't use the RocksDB caching (like in the older version) to see if the problem persist. Looking your account you have a new profile created less than a month ago that is only screaming against Bytecoin, maybe I'm wrong but honestly looks like a troll account.

**Comment by lollisoft** [comment]
Hi,

I am very new to crypto coins at all, but searched around to find this code here - enabling me to create my own crypto currency. (Happy)

Navigated from a forked git project to the parent (this here), I saw some live in the development. But very fast found an issue in the code that alerted me.

Assuming a server process or a daemon listening to incoming transactions or the like communication, I thought the code operates with multiple threads - eg - doing many transactions in the same time.

Then I searched for that and found some concrete hints in the code to really operating with threafs.

The problem: The code uses a non thread save flag ndicating a yet running 'thing'. But doesn't make the change of this flag thread save.

This may lead to some issues. I'll investigate some more time to prove my assumption and then I'll give some feedback. If a developer reads this, it is a change in one of the last commits where I have seen the code.

Regards,

Lothar

**Comment by lollisoft** [comment]
Just to hint, it is the mainContext.inExecutionQueue flag. And exactly the queueing code uses thread save code, but not the change of the flag. The code was introduced in the change of the following commit https://github.com/amjuarez/bytecoin/commit/1d48dc274048787b7a95814e216c61a1a7307ef1

**Comment by pmitchev** [comment]
Hello,
Do you have fix for this issue?

**Comment by keylength** [comment]
@lollisoft yes, can you please provide some more detail? Can you propose a fix? I've taken a look and everything seems intentional.

**Comment by keylength** [comment]
@lollisoft also please create a new thread because this one needs to be closed smh xD

**Comment by oldmankit** [comment]
I agree with @SMH17 It is clear that the OP is having serious issues, but the title of this issue is misleading. I can understand the desire to make a stink when your coins are in danger, but sticking to facts rather than semi-facts is probably the way to go.

For better of for worse, I just checked and changer.com seem to have re-enabled Bytecoin.

**Comment by lollisoft** [comment]
Hi,

please answer in git issue tracker. I do not yet have a solution. I have only taken a quick review, how the project is going and watched into the last checkin.

I need to get more details, how the code works. It seems that queued transactions are worked on in multiple threads. If the two treads should run both into the
blocking code, then there may be a possible race condition - I think. This is due to the fact that checking the blocking variable and setting it is not programmed in
a thread save manner.

You should understand this. You probably have more of such issues and I cannot review the complete code - in that short time.

Please think about the issue and what would happen if that becomes true. There are complaints in the issue tracker that may have to do with that code, but you should
be able to check for that.

Ao at the end I cannot give you more than using a mutex, or better a semaphore to block the code against executing it from different threads and then check and set the
real blocking variable.

Not doing so can cause real head-age :-)

I am not aware of the boost class library, but they should provide a semaphore implementation. A mutex is more expensive in computation and therefore less performant.
You only need mutex in cross process synchronization demands.

HTH,

Lothar

PS. I monitor the issue tracker. So communicate there (maybe in a new thread), is better. Users would appreciate this :-)

Am 21.06.2017 um 03:30 schrieb pmitchev:

**Comment by Sam-GI** [comment]
@Ben19904 Is this resolved yet?

**Comment by Ben19904** [comment]
Sorry I have been very busy lately with a new job so have not had time to check; @Sam-GI why would there be a possibility of it being resolved? It looks to me like no one has proposed a solution yet. Also with the code issue proposed by @lollisoft  it seems to me that he is talking about something in regards to this commit: https://github.com/amjuarez/bytecoin/commit/1d48dc274048787b7a95814e216c61a1a7307ef1 (i.e. version 2.0.6), however, my issue has been happening for 1.1.1, 1.1.4, and 1.1.7. The problem exists for the reference client too.

If the issue really is specifically down to the bytecoin wallet, then it must be in how it creates/modifies the .wallet file (in app data), and this must have existed at least since 1.1.1.

**Comment by Ben19904** [comment]
sorry I just realised that the commit link is for the reference client - the problem first started appearing for bytecoin wallet - not the ref client.

My problem can't have anything to do with code issues with the ref client.

**Comment by Sam-GI** [comment]
@Ben19904 "why would there be a possibility of it being resolved? It looks to me like no one has proposed a solution yet."

Simply because there are multiple communication channels. Anyhow, I'm in the same boa: copied address in BytecoinWallet, BCN were sent to it and a different address appeared in BytecoinWallet. No sign of the BCN or previous address …

**Comment by jplakali** [comment]
i transferred some bytecoin from cryptonator exchange to bytecoin desktop wallet and trying to sync the wallet but its syncing at very slow speed. i changed the node to 8080 and my wallet synced in 5 minutes but my wallet balance was still 0 so i changed the node to automatic and it started syncing again from 300k blocks. is there any way to take BCN out of the wallet without syncing the wallet? i can see confirmed transaction on chainradar. please help me.

**Comment by progietheus** [comment]
@jplakali Did you ever get this resolved? Having the same issue...It has been over a month, 0.00 BCN in Desktop wallet, but the pool shows transactions of me being paid.
